dhctl/pkg/operations/bootstrap/steps.go

```diff
@@ -120,71 +105,178 @@ func PrepareBashibleBundle(bundleName, nodeIP, devicePath string, metaConfig *co
+			Sudo().
+			WithTimeout(10 * time.Second)
+
+		if err := cmd.Run(); err != nil {
+			isReady = false
+			return err
+		}
+
+		stdout := string(cmd.StdoutBytes())
+		log.DebugF("cat %s stdout: '%s'\n", DHCTLEndBootstrapBashiblePipeline, stdout)
+
+		isReady = strings.TrimSpace(stdout) == "OK"
+
 		return nil
 	})
-}
 
-const (
-	bashibleInstalledMessage = `Bashible is already installed and healthy!
-	%s
-`
-	bashibleIsNotReadyMessage = `Bashible is not ready! Let's try to install it ...
-	Reason: %s
-`
-)
+	return isReady, err
+}
 
-func CheckBashibleBundle(sshClient *ssh.Client) bool {
-	var bashibleUpToDate bool
-	_ = log.Process("bootstrap", "Check Bashible", func() error {
-		bashibleCmd := sshClient.Command("/var/lib/bashible/bashible.sh", "--local").
-			Sudo().WithTimeout(20 * time.Second)
-		var output string
-
-		err := bashibleCmd.WithStdoutHandler(func(l string) {
-			if output == "" {
-				output = l
-			} else {
-				return
+func getBashiblePIDs(sshClient *ssh.Client) ([]string, error) {
+	var psStrings []string
+	h := func(l string) {
+		psStrings = append(psStrings, l)
+	}
+	cmd := sshClient.Command("ps", "a", "--no-headers", "-o", "args:64", "-o", "\"|%p\"").
+		Sudo().
+		WithTimeout(10 * time.Second).
+		WithStdoutHandler(h)
+
+	if err := cmd.Run(); err != nil {
+		var ee *exec.ExitError
+		// ssh exits with the exit status of the remote command or with 255 if an error occurred.
+		if errors.As(err, &ee) {
+			log.DebugF("'ps a --no-headers -o args:64 -o \"|%p\"' got exit code: %d and stderr %s", ee.ExitCode(), string(ee.Stderr))
+			if ee.ExitCode() == 255 {
+				return nil, err
 			}
-			switch {
-			case strings.Contains(output, "Can't acquire lockfile /var/lock/bashible."):
-				fallthrough
-			case strings.Contains(output, "Configuration is in sync, nothing to do."):
-				log.InfoF(bashibleInstalledMessage, output)
-				bashibleUpToDate = true
-			default:
-				log.InfoF(bashibleIsNotReadyMessage, output)
+
+			return nil, err
+		}
+	}
+
+	var res []string
+	for _, l := range psStrings {
+		log.DebugF("ps string: '%s'\n", l)
+
+		parts := strings.SplitN(l, "|", 2)
+		if len(parts) < 2 {
+			log.DebugLn("Skip ps string without pid")
+			continue
+		}
+
+		if !strings.Contains(parts[0], "bashible.sh") {
+			continue
+		}
+
+		pid := strings.TrimSpace(parts[1])
+		log.DebugF("Found bashible PID: %s\n", pid)
+
+		res = append(res, pid)
+	}
+
+	return res, nil
+}
+
+func killBashible(sshClient *ssh.Client, pids []string) error {
+	cmd := sshClient.Command("kill", pids...).
+		Sudo().
+		WithTimeout(10 * time.Second)
+
+	if err := cmd.Run(); err != nil {
+		var ee *exec.ExitError
+		// ssh exits with the exit status of the remote command or with 255 if an error occurred.
+		if errors.As(err, &ee) {
+			log.DebugF("'kill %v' got exit code: %d and stderr %s", pids, ee.ExitCode(), string(ee.Stderr))
+			if ee.ExitCode() == 255 {
+				return err
 			}
-		}).Run()
+
+			return nil
```

Comment:
I don't get it, we say it's okay even if bashible was not killed successfully?

Reply:
yep